lib/model/folder_sendrecv.go

```diff
@@ -1904,10 +1927,10 @@ func (f *sendReceiveFolder) deleteDirOnDisk(dir string, snap *db.Snapshot, scanC
 // scanIfItemChanged schedules the given file for scanning and returns errModified
 // if it differs from the information in the database. Returns nil if the file has
 // not changed.
-func (f *sendReceiveFolder) scanIfItemChanged(stat fs.FileInfo, item protocol.FileInfo, hasItem bool, scanChan chan<- string) (err error) {
+func (f *sendReceiveFolder) scanIfItemChanged(name string, stat fs.FileInfo, item protocol.FileInfo, hasItem bool, scanChan chan<- string) (err error) {
 	defer func() {
 		if err == errModified {
-			scanChan <- item.Name
+			scanChan <- name
```

Comment:
could have used stat.Name() instead of a new parameter? no big deal

Reply:
Not quite, because `stat.Name() == filepath.Base(name)`. Probably we should be consistent with using `name` for the last component and `path` otherwise - but that would be a pretty big cleanup job, because we aren't consistent at all on that o.O